nativelib/src/main/scala/scala/scalanative/runtime/ExecutionContext.scala

```diff
@@ -14,15 +14,27 @@ object ExecutionContext {
 
   private val queue: ListBuffer[Runnable] = new ListBuffer
 
+  @inline
+  private def loopRunOnceUnsafe(): Unit = {
+    val runnable = queue.remove(0)
+    try {
+      runnable.run()
+    } catch {
+      case t: Throwable =>
+        QueueExecutionContext.reportFailure(t)
+    }
+  }
+
   private[runtime] def loop(): Unit = {
     while (queue.nonEmpty) {
-      val runnable = queue.remove(0)
-      try {
-        runnable.run()
-      } catch {
-        case t: Throwable =>
-          QueueExecutionContext.reportFailure(t)
-      }
+      loopRunOnceUnsafe()
+    }
+  }
+
+  private[runtime] def loopRunOnce(): Int = {
```

Comment:
This method really doesn't loop at all. Consider calling it `runOne()` or perhaps `processOne()` or `executeOne()`.